components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java

```diff
@@ -164,6 +166,26 @@ public Response importApplicationsPost(InputStream fileInputStream, Attachment f
         return null;
     }
 
+    /**
+     * This extracts information for creating an APIKey from an OAuthApplication
+     * @param type Type of the OAuthApp(SANDBOX or PRODUCTION)
+     * @param oAuthApplicationInfo OAuth Application information
+     * @return An APIKey containing keys from OAuthApplication
+     */
+    private APIKey getAPIKeyFromOauthApp(String type, OAuthApplicationInfo oAuthApplicationInfo){
+        APIKey apiKey = new APIKey();
+        apiKey.setType(type);
+        apiKey.setConsumerKey(oAuthApplicationInfo.getClientId());
+        apiKey.setConsumerSecret(oAuthApplicationInfo.getClientSecret());
+        apiKey.setGrantTypes((String) oAuthApplicationInfo.getParameter("grant_types"));
+        if (apiKey.getGrantTypes().contains("implicit") && apiKey.getGrantTypes().contains("code")){
+            apiKey.setCallbackUrl((String) oAuthApplicationInfo.getParameter("redirect_uris"));
+        }
+        apiKey.setValidityPeriod(3600);
+        apiKey.setTokenScope("am_application_scope default");
+        return apiKey;
```

Comment:
Do we have to hard code these values

Reply:
As an offline discussion with @harsha89 it was decided to put this value 3600